include/persistent_data.hpp

```diff
@@ -190,6 +192,60 @@ class ConfigFile
         }
     }
 
+#ifdef BMCWEB_ENABLE_IBM_MANAGEMENT_CONSOLE
+    void writeCurrentSessionData()
+    {
+        std::ofstream persistentFile(dumpFilename);
+        // set the permission of the file to 640
+        std::filesystem::perms permission =
+            std::filesystem::perms::owner_read |
+            std::filesystem::perms::owner_write |
+            std::filesystem::perms::group_read;
+        std::filesystem::permissions(filename, permission);
+        const auto& c = SessionStore::getInstance().getAuthMethodsConfig();
+        const auto& eventServiceConfig =
+            EventServiceStore::getInstance().getEventServiceConfig();
+        nlohmann::json data{
+            {"auth_config",
+             {{"XToken", c.xtoken},
+              {"Cookie", c.cookie},
+              {"SessionToken", c.sessionToken},
+              {"BasicAuth", c.basic},
+              {"TLS", c.tls}}
+
+            },
+            {"eventservice_config",
+             {{"ServiceEnabled", eventServiceConfig.enabled},
+              {"DeliveryRetryAttempts", eventServiceConfig.retryAttempts},
+              {"DeliveryRetryIntervalSeconds",
+               eventServiceConfig.retryTimeoutInterval}}
+
+            },
+            {"system_uuid", systemUuid},
+            {"revision", jsonRevision},
+            {"timeout", SessionStore::getInstance().getTimeoutInSeconds()}};
+
+        nlohmann::json& sessions = data["sessions"];
+        sessions = nlohmann::json::array();
+        for (const auto& p : SessionStore::getInstance().authTokens)
+        {
+            if (p.second->persistence !=
+                persistent_data::PersistenceType::SINGLE_REQUEST)
+            {
+                sessions.push_back({
+                    {"unique_id", p.second->uniqueId},
+                    {"session_token", p.second->sessionToken},
+                    {"username", p.second->username},
+                    {"csrf_token", p.second->csrfToken},
+                    {"client_ip", p.second->clientIp},
+                    {"client_id", p.second->clientId},
+                });
+            }
+        }
+        persistentFile << data;
+    }
```

Comment:
I am curious what happens if there is a problem writing the dump file.  (It looks like this code was just copied from the writeData() function below.)  If the file cannot be written, is an error indicated?

Reply:
I see any  need to log error in this case of failure

Reply:
If it this important of information seems like we would want a log knowing it failed to write